### PR TITLE
New pfp

### PR DESCRIPTION
**Pull request by cccyyyr**
Refactored activity class and profile pic branch is ready.

### CODE REVIEW

**Comment by crchong1** [comment]
@cccyyyr can you fix the merge conflicts

File: src/main/User/UserController.java [hunk 1]
```diff
@@ -271,4 +279,35 @@ public static JSONObject mergeJSON(
...
+
+  public Handler uploadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        UploadedFile file = ctx.uploadedFile("file");
+        logger.info(username + " is attempting to upload a profile picture");
+        UploadPfpService serv = new UploadPfpService(db, logger, username, file);
+        JSONObject res = serv.executeAndGetResponse().toJSON();
+        logger.info(username + " has successfully uploaded a profile picture");
+        ctx.json(res.toString());
```
Comment (crchong1) [inline src/main/User/UserController.java:302]: you should try not to use ctx.json. You can just do: ctx.result(res.toResponseString()) i think

File: src/main/User/UserController.java [hunk 2]
```diff
@@ -271,4 +279,35 @@ public static JSONObject mergeJSON(
     }
     return merged;
   }
+
+  public InputStream getUserPfp(String username) {
```
Comment (crchong1) [inline src/main/User/UserController.java:283]: can we move this into a service and not let this sit in the usercontroller? it just makes it hard to test later on

File: src/main/User/UserController.java [hunk 3]
```diff
@@ -271,4 +279,35 @@ public static JSONObject mergeJSON(
...
+        JSONObject res = serv.executeAndGetResponse().toJSON();
+        logger.info(username + " has successfully uploaded a profile picture");
+        ctx.json(res.toString());
+      };
+
+  public Handler loadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        LoadPfpService lps = new LoadPfpService(db, logger, username);
+        JSONObject mes = lps.executeAndGetResponse().toJSON();
```
Comment (crchong1) [inline src/main/User/UserController.java:309]: are you using this message anywhere? it doesn't seem like if there is an error code, anything gets handled

File: src/main/User/UserController.java [hunk 4]
```diff
@@ -271,4 +279,35 @@ public static JSONObject mergeJSON(
...
+        ctx.header("Content-Type", "application/pfp");
```
Comment (crchong1) [inline src/main/User/UserController.java:310]: also i dont think this is a correct content-type header. see: https://www.geeksforgeeks.org/http-headers-content-type/

**Review by crchong1 — CHANGES_REQUESTED** [review]

File: src/test/UserTest/PfpTestIntegration.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    assert (get.isSuccess());
+  }
+
+  public static void getPfp() {
+    File examplePDF = new File(currentPfpFolderPath + File.separator + "mvc.png");
+    HttpResponse<String> uploadResponse =
+        Unirest.post(TestUtils.getServerUrl() + "/upload-pfp")
+            .header("Content-Disposition", "attachment")
+            .field("file", examplePDF)
+            .asString();
```
Comment (crchong1) [inline src/test/UserTest/PfpTestIntegration.java:69]: maybe also assert that the uploadResponse is success too?

**Comment by crchong1** [comment]
also can you remove your DS_STORE file

File: src/main/User/UserController.java (hunk 2 above)
Comment (cccyyyr) [reply src/main/User/UserController.java:283]: it was never used ooops.  I already moved to a service. I just deleted it.

File: src/main/Config/AppConfig.java [hunk 6]
```diff
@@ -84,7 +85,7 @@ public static Javalin appFactory(DeploymentLevel deploymentLevel) {
     app.post("/change-two-factor-setting", accountSecurityController.change2FASetting);
 
     /* -------------- SUBMIT BUG------------------ */
-    app.post("/submit-issue", issueController.submitIssue);
+    //    app.post("/submit-issue", issueController.submitIssue);
```
Comment (crchong1) [inline src/main/Config/AppConfig.java:88]: is there a reason why we are commenting out submit-issue?

File: src/main/User/Services/LoadPfpService.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSFile grid_out = gridBucket.find(filter).limit(1).first();
+    if (grid_out == null || grid_out.getMetadata() == null) {
+      return null;
+    }
+    String fileName = grid_out.getFilename();
+    String[] temp = fileName.split("\\.");
+    contentType = temp[temp.length - 1];
+    if (!contentType.equals("png")) {
+      contentType = "jpeg";
```
Comment (crchong1) [inline src/main/User/Services/LoadPfpService.java:42]: are we only accepting png and jpeg? Can we parse the content type from the file?

**Review by crchong1 — CHANGES_REQUESTED** [review]

**Review by crchong1 — APPROVED** [review]
can you remove DS_STORE from the diff? other than that approved

**Pull request merged**